src/file/format/jsonc.rs

```diff
@@ -0,0 +1,61 @@
+use std::error::Error;
+
+use crate::error::{ConfigError, Unexpected};
+use crate::map::Map;
+use crate::value::{Value, ValueKind};
+
+use jsonc_parser::JsonValue;
+
+pub(crate) fn parse(
+    uri: Option<&String>,
+    text: &str,
+) -> Result<Map<String, Value>, Box<dyn Error + Send + Sync>> {
+    match jsonc_parser::parse_to_value(text, &Default::default())? {
+        Some(r) => match r {
+            JsonValue::String(ref value) => Err(Unexpected::Str(value.to_string())),
+            JsonValue::Number(value) => Err(Unexpected::Float(value.parse::<f64>().unwrap())),
+            JsonValue::Boolean(value) => Err(Unexpected::Bool(value)),
+            JsonValue::Object(o) => match from_jsonc_value(uri, JsonValue::Object(o)).kind {
+                ValueKind::Table(map) => Ok(map),
+                _ => unreachable!(),
+            },
+            JsonValue::Array(_) => Err(Unexpected::Seq),
+            JsonValue::Null => Err(Unexpected::Unit),
+        },
+        None => Err(Unexpected::Unit),
+    }
+    .map_err(|err| ConfigError::invalid_root(uri, err))
+    .map_err(|err| Box::new(err) as Box<dyn Error + Send + Sync>)
```

Comment:
This suggestion will require adding an import: `use crate::format;`

```suggestion
    let value = from_jsonc_value(uri, jsonc_parser::parse_to_value(text, &Default::default())?);
    format::extract_root_table(uri, value)
```